src/signer.js

```diff
@@ -102,7 +102,7 @@ function sign(
     ...fixedPayload,
     iat: noTimestamp ? undefined : Math.floor(iat / 1000),
     exp: payload.exp ? payload.exp : expiresIn ? Math.floor((iat + expiresIn) / 1000) : undefined,
-    nbf: notBefore ? Math.floor((iat + notBefore) / 1000) : undefined
+    nbf: payload.nbf ? payload.nbf : notBefore ? Math.floor((iat + notBefore) / 1000) : undefined
```

Comment:
I think it would be easier to simply put the `...payload, ...fixedPayload` at the end of the object, allowing the payload to override `iat`, `exp`, and `nbf`. This would eliminate the need for the repetitive `payload.nbf ? payload.nbf :` checks, as well as fix an edge case where `payload.nbf` is present but falsy.

```js
const finalPayload = {
    iat: noTimestamp ? undefined : Math.floor(iat / 1000),
    exp: expiresIn ? Math.floor((iat + expiresIn) / 1000) : undefined,
    nbf: notBefore ? Math.floor((iat + notBefore) / 1000) : undefined,
    ...payload,
    ...fixedPayload,
}
```

Reply:
Good point, I will.

Reply:
There is a small inconsistency of behavior when payload.exp is NaN. Instead of ignoring the field the serialization produces null.
```
JSON.stringify({ a: Number.NaN }) // '{"a":null}'
```
So if we want to keep the behavior consistent with before we have to add an additional check on payload.exp making the code less readable defeating, in my opinion, the purpose of a changes like this one. (same issue with nbf). So I will keep the implementation as is.

Reply:
Fair enough. Looking at the code, `payload.iat` seems to be ignored as well, so perhaps the same change should be applied to that one as well?

Reply:
It is not, there is specific handling for that case. See line 98.
```
 const iat = payload.iat * 1000 || clockTimestamp || Date.now()
```
There also tests that are specific to this use case.